instrumentation/github.com/emicklei/go-restful/otelrestful/restful.go

```diff
@@ -84,3 +86,11 @@ func OTelFilter(service string, opts ...Option) restful.FilterFunction {
 		}
 	}
 }
+
+// RouteTaggerMiddleware annotates the current span with the handler's route path.
+func RouteTaggerMiddleware() restful.FilterFunction {
+	return func(request *restful.Request, response *restful.Response, chain *restful.FilterChain) {
+		otelhttp.AnnotateSpanWithHTTPRoute(request.Request.Context(), request.SelectedRoutePath())
+		chain.ProcessFilter(request, response)
+	}
+}
```

Comment:
We should not depend on `otelhttp` in other instrumentation libraries. These libraries are not stable and bumping in `otelhttp` can break `otelgin` users. The `AnnotateSpanWithHTTPRoute` implementation is so small that I would rather inline it in existing instrumentation packages.

I would prefer if this PR is only about `otelhttp`.

I would move the middleware implementations using `AnnotateSpanWithHTTPRoute` to examples under `otelhttp`. This would demonstrate how someone could use `otelhttp` e.g. instead of `otelmux`.